src/cascadia/TerminalSettingsEditor/Actions.cpp

```diff
@@ -4,61 +4,261 @@
+        for (const auto& kbdVM : _KeyBindingList)
+        {
+            get_self<KeyBindingViewModel>(kbdVM)->IsAutomationPeerAttached(_AutomationPeerAttached);
+        }
+        return nullptr;
     }
 
     void Actions::OnNavigatedTo(const NavigationEventArgs& e)
     {
         _State = e.Parameter().as<Editor::ActionsPageNavigationState>();
 
-        std::vector<Command> keyBindingList;
-        for (const auto& [_, command] : _State.Settings().GlobalSettings().ActionMap().NameMap())
+        // Convert the key bindings from our settings into a view model representation
+        const auto& keyBindingMap{ _State.Settings().ActionMap().KeyBindings() };
+        std::vector<Editor::KeyBindingViewModel> keyBindingList;
+        keyBindingList.reserve(keyBindingMap.Size());
+        for (const auto& [keys, cmd] : keyBindingMap)
+        {
+            auto container{ make_self<KeyBindingViewModel>(keys, cmd) };
+            container->PropertyChanged({ this, &Actions::_ViewModelPropertyChangedHandler });
+            container->DeleteKeyBindingRequested({ this, &Actions::_ViewModelDeleteKeyBindingHandler });
+            container->RebindKeysRequested({ this, &Actions::_ViewModelRebindKeysHandler });
+            container->IsAutomationPeerAttached(_AutomationPeerAttached);
+            keyBindingList.push_back(*container);
+        }
+
+        std::sort(begin(keyBindingList), end(keyBindingList), KeyBindingViewModelComparator{});
+        _KeyBindingList = single_threaded_observable_vector(std::move(keyBindingList));
+    }
+
+    void Actions::_ViewModelPropertyChangedHandler(const IInspectable& sender, const Windows::UI::Xaml::Data::PropertyChangedEventArgs& args)
+    {
+        const auto senderVM{ sender.as<Editor::KeyBindingViewModel>() };
+        const auto propertyName{ args.PropertyName() };
+        if (propertyName == L"IsInEditMode")
         {
-            // Filter out nested commands, and commands that aren't bound to a
-            // key. This page is currently just for displaying the actions that
-            // _are_ bound to keys.
-            if (command.HasNestedCommands() || !command.Keys())
+            if (senderVM.IsInEditMode())
+            {
+                for (uint32_t i = 0; i < _KeyBindingList.Size(); ++i)
+                {
+                    const auto& kbdVM{ _KeyBindingList.GetAt(i) };
+                    if (senderVM == kbdVM)
+                    {
+                        // move focus to the edit mode controls
+                        const auto& container{ KeyBindingsListView().ContainerFromIndex(i).try_as<ListViewItem>() };
+                        container.Focus(FocusState::Programmatic);
+                    }
+                    else
+                    {
+                        // exit edit mode for all other containers
+                        get_self<KeyBindingViewModel>(kbdVM)->DisableEditMode();
+                    }
+                }
+            }

```

Comment:
This whole section definitely needs some comments throughout explaining what is going on here, this is hard to follow